Fix X509 load locations to handle PEM files with multiple certs

ColtonWilley opened the pull request:
# Description

wolfSSL_X509_load_locations() previously did not handle PEM files with multiple certs in one file, only the first cert would be loaded. Modify wolfSSL_X509_load_locations() to fix this by using existing logic in x509.c to loop over the PEM headers and load each item, cert or CRL. 

Fixes zd 19835

# Checklist

 - [x] added tests

ColtonWilley commented:
Retest this please.

julek-wolfssl commented on tests/api.c:
```diff
@@ -28968,6 +28968,39 @@ static int test_wolfSSL_X509_STORE_load_locations(void)
     return EXPECT_RESULT();
 }
 
+static int test_wolfSSL_X509_STORE_load_multiple_certs(void)
```
We don't have to move all tests at once. We can move them one by one and make sure that new tests are written there. We want to stop the growth of `api.c`. It already takes long enough to compile.

ColtonWilley replied:
Tests moved to test_x509.c

julek-wolfssl reviewed (dismissed):
Please move the new tests and check if the CRL functionality is tested somewhere.

ColtonWilley commented on tests/api.c:
```diff
@@ -28968,6 +28968,39 @@ static int test_wolfSSL_X509_STORE_load_locations(void)
+        if (obj && X509_OBJECT_get_type(obj) == X509_LU_X509) {
+            cert_count++;
+        }
+    }
```
Subject name check added.

julek-wolfssl commented:
@ColtonWilley please rebase to run all the latest tests.

ColtonWilley committed:
Fix X509 load locations to handle PEM files with multiple certs

Adds X509LoadPemFile to walk multi-cert PEM files when loading via
wolfSSL_X509_STORE_load_locations and X509_LOOKUP_load_file, replacing
the single-cert helpers X509StoreReadFile/X509StoreLoadFile which only
read the first cert from a file.

Rebased fresh onto current upstream master (was 2209 commits behind);
test additions deferred to a follow-up since the test file layout has
been reorganized in master.
